src/ome_zarr_models/_rfc5_transforms/coordinate_transformations.py

```diff
@@ -0,0 +1,93 @@
+from __future__ import annotations
+
+from typing import Annotated, Literal
+
+from pydantic import Field, field_validator
+
+from ome_zarr_models._rfc5_transforms.axes import Axes
+from ome_zarr_models._utils import duplicates
+from ome_zarr_models.base import BaseAttrs
+
+
+class CoordinateSystem(BaseAttrs):
+    name: str
+    axes: Axes
+
+    @field_validator("name")
+    def name_must_not_be_empty(cls, value: str) -> str:
+        if value == "":
+            raise ValueError("name must be a non-empty string")
+        return value
+
+    @field_validator("axes")
+    def axes_must_not_be_empty(cls, value: Axes) -> Axes:
+        if len(value) == 0:
+            raise ValueError("axes must contain at least one axis")
+        return value
+
+    @field_validator("axes", mode="after")
+    @classmethod
+    def _ensure_unique_axis_names(cls, axes: Axes) -> Axes:
```

Comment:
There's already `ome_zarr_models.common.validation.unique_items_validator` that I think would work here.

Reply:
Thanks! Simplified.